src/sentry/api/endpoints/notifications/notification_actions_details.py

```diff
@@ -0,0 +1,60 @@
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization_flag import FlaggedOrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.notification_action import NotificationActionSerializer
+from sentry.models.notificationaction import NotificationAction
+from sentry.models.organization import Organization
+
+
+@region_silo_endpoint
+class NotificationActionsDetailsEndpoint(FlaggedOrganizationEndpoint):
+    feature_flags = ["organizations:notification-actions"]
+
+    def get(self, request: Request, organization: Organization, action_id: int) -> Response:
+        try:
+            action = NotificationAction.objects.get(
+                id=action_id,
+                organization_id=organization.id,
+            )
+        except NotificationAction.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
this try/except clause is in every method in this file. also, you might want to raise `ResourceDoesNotExist`

Reply:
You can also implement `convert_args()` to handle record lookup with common failure actions. You could convert `action_id` into the notification action and add `action` to kwargs.

Reply:
Makes sense, pulled it into a `convert_args` call instead